src/it/resources/com/google/checkstyle/test/chapter2filebasic/rule231filetab/InputWhitespaceCharacters.java

```diff
@@ -5,194 +5,51 @@ final class InputWhitespaceCharacters {
-    // a line
-    // a line
-    // a line
-  }
-
-  /** test local variables */
-  private void localVariables() {
-    int abc = 0;
-    int ABC = 0;
-
-    final int cde = 0;
-    final int CDE = 0;
-
-    for (int k = 0; k < 1; k++) {
-      String innerBlockVariable = "";
-    }
-    for (int I = 0; I < 1; I++) {
-      String InnerBlockVariable = "";
-    }
-  }
-
-  /** test method pattern */
-  void ALL_UPPERCASE_METHOD() {}
-
-  /** test illegal constant * */
-  private static final int BAD__NAME = 3;
-
   // A very, very long line that is OK because it matches the regexp "^.*is OK.*regexp.*$"
   // long line that has a tab ->	<- and would be OK if tab counted as 1 char
   // violation above 'Line contains a tab character.'
   // tabs that count as one char because of their position ->	<-   ->	<-
   // violation above 'Line contains a tab character.'
 
-  /** some lines to test the column after tabs */
+  /** some lines to test the column after tabs. */
   void violateColumnAfterTabs() {
     // with tab-width 8 all statements below start at the same column,
     // with different combinations of ' ' and '\t' before the statement
-    int tab0 =1;
-  	int tab1 =1; // violation 'Line contains a tab character.'
-  	int tab2 =1; // violation 'Line contains a tab character.'
-		int tab3 =1; // violation 'Line contains a tab character.'
- 	 	int tab4 =1; // violation 'Line contains a tab character.'
-  	int tab5 =1; // violation 'Line contains a tab character.'
-  }
-
-  // MEMME:
-  /* MEMME: a
-   * MEMME:
-   * OOOO
-   */
-  /* NOTHING */
-  /* YES */
-  /* MEMME: x */
-  /* YES!! */
-
-  /** test long comments * */
-  void veryLong() {
-    /*
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    blah blah blah blah
-    enough talk */
+    int tab0 = 1;
+  	int tab1 = 1;
+   // 2 violations above:
+   //  'Line contains a tab character.'
+   //  ''method def' child has incorrect indentation level 8, expected level should be 4.'
+  	int tab2 = 1;
+  // 2 violations above:
+  //  'Line contains a tab character.'
+  //  ''method def' child has incorrect indentation level 8, expected level should be 4.'
+		int tab3 = 1;
+    // 2 violations above:
+    //  'Line contains a tab character.'
+    //  ''method def' child has incorrect indentation level 16, expected level should be 4.'
+ 	 	int tab4 = 1;
+   // 2 violations above:
+   //  'Line contains a tab character.'
+   //  ''method def' child has incorrect indentation level 16, expected level should be 4.'
+  	int tab5 = 1;
+   // 2 violations above:
+   //  'Line contains a tab character.'
+   //  ''method def' child has incorrect indentation level 8, expected level should be 4.'
```

Comment:
indentation for violation comments became weird because of the tabs used in their preceded lines.

Reply:
Ok, if target message is one of few on same line, ok to keep it.